prime-router/src/main/kotlin/Translator.kt

```diff
@@ -133,16 +133,20 @@ class Translator(private val metadata: Metadata, private val settings: SettingsP
         )
         if (routingFilteredReport.isEmpty()) return routingFilteredReport
 
-        // Do conditionFiltering on the routingFilteredReport
-        val conditionFilteredReport = filterByOneFilterType(
-            routingFilteredReport,
-            receiver,
-            organization,
-            ReportStreamFilterType.CONDITION_FILTER,
-            trackingElement,
-            doLogging = true
-        )
-        if (conditionFilteredReport.isEmpty()) return conditionFilteredReport
+        var conditionFilteredReport = routingFilteredReport
+        // condition filter can only be used on UP
+        if (receiver.topic == Topic.FULL_ELR || receiver.topic == Topic.FULL_ELR_TEST) {
+            // Do conditionFiltering on the routingFilteredReport
+            conditionFilteredReport = filterByOneFilterType(
+                routingFilteredReport,
+                receiver,
+                organization,
+                ReportStreamFilterType.CONDITION_FILTER,
+                trackingElement,
+                doLogging = true
+            )
+            if (conditionFilteredReport.isEmpty()) return conditionFilteredReport
+        }
```

Comment:
Is the the return on this if statement correct? If conditionFilteredReport is empty it's just going to return the conditionFilteredReport

Reply:
Yes, just as with the other filters, if all have been filtered out then there is no more processing to do and we can just return.